deposit-app/src/main/java/edu/unc/lib/boxc/deposit/fcrepo4/IngestContentObjectsJob.java

```diff
@@ -376,64 +376,62 @@ private FileObject addFileToWork(WorkObject work, Resource childResc)
-                        throw new JobInterruptedException(e.getMessage());
+        if (originalResc != null) {
+            String storageString = getPropertyValue(originalResc, CdrDeposit.storageUri);
+            URI storageUri = URI.create(storageString);
+            // Pull out file properties if they are present
+            String mimetype = getPropertyValue(originalResc, CdrDeposit.mimetype);
+
+            String sha1 = getPropertyValue(originalResc, CdrDeposit.sha1sum);
+            String md5 = getPropertyValue(originalResc, CdrDeposit.md5sum);
+
+            // Label comes from file object if not set for binary
+            String label = getPropertyValue(originalResc, CdrDeposit.label);
+            if (label == null) {
+                label = getPropertyValue(childResc, CdrDeposit.label);
+            }
+
+            // Add the file to the work as the datafile of its own FileObject
+            // Retry if there are checksum failures
+            for (int retryCnt = 1; retryCnt <= CHECKSUM_RETRIES; retryCnt++) {
+                try {
+                    fileObj = work.addDataFile(childPid, storageUri, label, mimetype, sha1, md5, aipModel);
+                    break;
+                } catch (ChecksumMismatchException e) {
+                    if ((CHECKSUM_RETRIES - retryCnt) > 0) {
+                        log.warn("Failed to ingest file {} due to a checksum mismatch, {} retries remaining: {}",
+                                childPid.getQualifiedId(), CHECKSUM_RETRIES - retryCnt, e.getMessage());
+                        try {
+                            Thread.sleep(retryCnt * 1000);
+                        } catch (InterruptedException ef) {
+                            throw new JobInterruptedException(e.getMessage());
+                        }
+                    } else {
+                        failJob("Unable to ingest " + childPid.getQualifiedId(), e.getMessage());
                     }
-                } else {
-                    failJob("Unable to ingest " + childPid.getQualifiedId(), e.getMessage());
                 }
             }
-        }
 
-        // Record the size of the file for throughput stats
-        if (storageUri.getScheme().equals("file")) {
-            metricsClient.incrDepositFileThroughput(getDepositUUID(), Paths.get(storageUri).toFile().length());
-        }
+            // Record the size of the file for throughput stats
+            if (storageUri.getScheme().equals("file")) {
+                metricsClient.incrDepositFileThroughput(getDepositUUID(), Paths.get(storageUri).toFile().length());
+            }
 
-        // Add the FITS report for this file
-        addFitsHistory(fileObj, childResc);
-        addFitsReport(fileObj, childResc);
+            // Add the FITS report for this file
+            addFitsHistory(fileObj, childResc);
+            addFitsReport(fileObj, childResc);
+        } else {
+            fileObj = repoObjFactory.createFileObject(childPid, aipModel);
```

Comment:
might be helpful to our future selves to add a comment here to indicate what scenario is in which we would want to be creating a fileObject without an original_file